File: 476/number-complement.go
package main

import "fmt"

func findComplement(num int) int {
	tmp := num
	res := 0
	for tmp > 0 {
		res = res*2 + 1
		tmp = tmp >> 1
	}

	return res ^ num
}

func main() {
	num := 5
	fmt.Println(findComplement(num))
}

File: 476/number-complement.py
class Solution:
    def findComplement(self, num):
        """
        :type num: int
        :rtype: int
        """
        tmp = num
        res = 0
        while tmp > 0:
            res = res * 2 + 1
            tmp = tmp >> 1
        
        res = res ^ num
        return res
